sql/src/main/scala/org/apache/spark/sql/sedona_sql/strategy/join/JoinQueryDetector.scala

```diff
@@ -69,6 +69,8 @@ class JoinQueryDetector(sparkSession: SparkSession) extends Strategy {
           Some(JoinQueryDetection(left, right, leftShape, rightShape, false, extraCondition))
         case ST_Crosses(Seq(leftShape, rightShape)) =>
           Some(JoinQueryDetection(right, left, rightShape, leftShape, false, extraCondition))
+        case ST_Disjoint(Seq(leftShape, rightShape)) =>
+          Some(JoinQueryDetection(left, right, leftShape, rightShape, false, extraCondition))
         case _ => None
```

Comment:
Please do NOT make any change to this file because ST_Disjoint-based join in Spark cannot be optimized bySedona.